src/engine/sparqlExpressions/NumericBinaryExpressions.cpp

```diff
@@ -13,18 +14,32 @@ NARY_EXPRESSION(MultiplyExpression, 2,
 
 // Division.
 //
-// TODO<joka921> If `b == 0` this is technically undefined behavior and
-// should lead to an expression error in SPARQL. Fix this as soon as we
-// introduce the proper semantics for expression errors.
-// Update: I checked it, and the standard differentiates between `xsd:decimal`
-// (error) and `xsd:float/xsd:double` where we have `NaN` and `inf` results. We
-// currently implement the latter behavior. Note: The result of a division in
+// TODO<joka921> If `b == 0` the the behavior of the division depends on whether
+// the inputs are `xsd:decimal` or `xsd:double` (`double`s have special values
+// like `NaN` or `infinity`, decimals don't). As we currently make no difference
+// between those two types, we have to choose one of the behaviors. We make the
+// result `UNDEF` in this case to pass the sparql conformance tests that rely on
+// this behavior.
+// Note: The result of a division in
 // SPARQL is always a decimal number, so there is no integer division.
+template <bool DivisionByZeroIsUndef>
 [[maybe_unused]] inline auto divideImpl = [](auto x, auto y) {
+  if constexpr (DivisionByZeroIsUndef) {
+    if (y == 0) {
+      return std::numeric_limits<double>::quiet_NaN();
+    }
```

Comment:
This is a little bit misleading, so please extend the comment explaining this:
`NaN` is currently converted to `UNDEF` outside of this function, so by explicitly returning `NaN` in this case, we force the engine to return `UNDEF` for these kinds of results, but it seems like `NaN` would be a valid representation for [`xsd:float`](https://www.w3.org/TR/xmlschema-2/#float) and [`xsd:double`](https://www.w3.org/TR/xmlschema-2/#double), so this conversion to `UNDEF` would be wrong. A fix of this behaviour would then break this code.